rust/ql/lib/codeql/rust/internal/PathResolution.qll

```diff
@@ -1687,6 +1711,14 @@ private ItemNode resolvePathCand(RelevantPath path) {
     or
     not pathUsesNamespace(path, _) and
     not path = any(MacroCall mc).getPath()
+  ) and
+  (
+    not path = CallExprImpl::getFunctionPath(_)
+    or
+    exists(CallExpr ce |
+      path = CallExprImpl::getFunctionPath(ce) and
+      result.(CallableItemNode).getNumberOfParameters() = ce.getNumberOfArgs()
+    )
```

Comment:
Is it correct that Rust's own path resolution specification doesn't rely on arities, and that this is about pruning mistakes in path resolution?